plugins/woocommerce-admin/client/wp-admin-scripts/command-palette/index.js

```diff
@@ -0,0 +1,164 @@
+			const query = {
+				search: !! search ? search : undefined,
+				per_page: 10,
+				orderby: search ? 'relevance' : 'date',
+				status: [ 'publish', 'future', 'draft', 'pending', 'private' ],
+			};
+			return {
+				records: getEntityRecords( 'postType', postType, query ),
+				isLoading: ! select( coreStore ).hasFinishedResolution(
+					'getEntityRecords',
+					[ 'postType', postType, query ]
+				),
+			};
+		},
+		[ search ]
+	);
+
+	const commands = useMemo( () => {
+		return ( records ?? [] ).map( ( record ) => {
+			const command = {
+				name: postType + '-' + record.id,
+				searchLabel: record.title?.rendered + ' ' + record.id,
+				label: record.title?.rendered
+					? record.title?.rendered
+					: __( '(no title)', 'woocommerce' ),
+				icon: box,
+			};
+			return {
+				...command,
+				callback: ( { close } ) => {
+					queueRecordEvent( 'woocommerce_command_palette_submit', {
+						name: 'woocommerce/product',
+					} );
+
+					const args = {
+						post: record.id,
+						action: 'edit',
+					};
+					const targetUrl = addQueryArgs( 'post.php', args );
+					document.location = targetUrl;
+					close();
+				},
+			};
+		} );
+	}, [ records ] );
+
+	return {
+		commands,
+		isLoading,
+	};
+}
+
+const WooCommerceCommands = () => {
+	useCommandWithTracking( {
+		name: 'woocommerce/add-new-product',
+		label: __( 'Add new product', 'woocommerce' ),
+		icon: plus,
+		callback: () => {
+			document.location = addQueryArgs( 'post-new.php', {
+				post_type: 'product',
+			} );
+		},
+	} );
+	useCommandWithTracking( {
+		name: 'woocommerce/add-new-order',
+		label: __( 'Add new order', 'woocommerce' ),
+		icon: plus,
+		callback: () => {
+			document.location = addQueryArgs( 'admin.php', {
+				page: 'wc-orders',
+				action: 'new',
+			} );
+		},
+	} );
+	useCommandWithTracking( {
+		name: 'woocommerce/view-products',
+		label: __( 'Products', 'woocommerce' ),
+		icon: box,
+		callback: () => {
+			document.location = addQueryArgs( 'edit.php', {
+				post_type: 'product',
+			} );
+		},
+	} );
+	useCommandWithTracking( {
+		name: 'woocommerce/view-orders',
+		label: __( 'Orders', 'woocommerce' ),
+		icon: box,
+		callback: () => {
+			document.location = addQueryArgs( 'admin.php', {
+				page: 'wc-orders',
+			} );
+		},
+	} );
+	useCommandLoader( {
+		name: 'woocommerce/product',
+		hook: useProductCommandLoader,
+	} );
+	useWooCommerceSettingsCommand( {
+		label: __( 'Products Settings', 'woocommerce' ),
+		tab: 'products',
+	} );
+	useWooCommerceSettingsCommand( {
+		label: __( 'Shipping Settings', 'woocommerce' ),
+		tab: 'shipping',
+	} );
+	useWooCommerceSettingsCommand( {
+		label: __( 'Payments Settings', 'woocommerce' ),
+		tab: 'checkout',
+	} );
+	useWooCommerceSettingsCommand( {
+		label: __( 'Accounts & Privacy Settings', 'woocommerce' ),
+		tab: 'account',
+	} );
+	useWooCommerceSettingsCommand( {
+		label: __( 'WooCommerce Emails Settings', 'woocommerce' ),
+		tab: 'email',
+	} );
```

Comment:
I wonder if we can make this dynamic, like getting the setting tab in the PHP side then passing the data to the editor? If we can do that, we can support custom setting tabs added by extensions. Of course we can work on that in later iterations.

Reply:
Yeah, great suggestion! :100: 

I implemented it in b7dd33abd0281af51fcabce695b96c28169bc66c. It's quite a big commit as I needed to refactor some parts of the PR. I also changed the command titles from `Products Settings` to `WooCommerce Settings: Products`, I think it would work better with extension tabs and other tabs with generic names. Same for the `WooCommerce Anayltics: ...` commands. Let me know how it looks!

Reply:
> I also changed the command titles from `Products Settings` to `WooCommerce Settings: Products`

Great move!